mcsema/cfgToLLVM/x86Instrs_String.cpp

```diff
@@ -206,6 +206,74 @@ static llvm::BasicBlock *doStosV(llvm::BasicBlock *pred) {
   return doWrite;
 }
 
+template<int opSize, int regWidth>
+static llvm::BasicBlock *doLodsV(llvm::BasicBlock *pred) {
+  //read [ESI]
+  auto srcRegVal = R_READ<regWidth>(pred, llvm::X86::RSI);
+  auto fromMem = M_READ_0<opSize>(pred, srcRegVal);
+
+  // Store [ESI] in EAX
+  R_WRITE<opSize>(pred, llvm::X86::RAX, fromMem);
+
+  //now, either increment or decrement ESI based on the DF flag
+  auto &C = pred->getContext();
+  auto F = pred->getParent();
+  auto isZero = llvm::BasicBlock::Create(C, "", F);
+  auto isOne = llvm::BasicBlock::Create(C, "", F);
+  auto doWrite = llvm::BasicBlock::Create(C, "", F);
+
+  //compare DF against 0
+  auto cmpRes = new llvm::ICmpInst( *pred, llvm::CmpInst::ICMP_EQ,
+                                   F_READ(pred, llvm::X86::DF),
+                                   CONST_V<1>(pred, 0), "");
+
+  //do a branch based on the cmp
+  llvm::BranchInst::Create(isZero, isOne, cmpRes, pred);
+
+  uint64_t disp = 0;
```

Comment:
Alternative, go add a utility function somewhere that encapsulates this type of check. I think they're done in various different places, would be nice to dedup.